Clarification of Zone and Surface Opaque Conduction Output Variables

RKStrand opened the pull request:
Pull request overview
---------------------

- Fixes #10812 

### Description of the purpose of this PR

Several output variables that were listed in the Input/Output Reference relating to zone and surface opaque conduction values were not actually set up to allow them to be produced.  A full review of these variables revealed that there were additional variables that were neither set-up for reporting in the code nor mentioned in the Input/Output Reference but were defined and calculated within the code for reporting purposes.  In addition, there was some naming inconsistencies noted in these variables as well.  Changes include making all variables that were set-up for reporting available by adding code to allow this, correcting and enhancing the Input/Output Reference for these variables, providing variable name change rules for four variables where the name was changed for clarify/consistency, an existing test file IDF was modified to request all of these variables, and a unit test to verify that these variables are being reported correctly.  The only output changes would be in the file that had additional variables requested.

### Pull Request Author

<!-- Add to this list or remove from it as applicable.  This is a simple templated set of guidelines. -->

 - [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [X] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer

<!-- This will not be exhaustively relevant to every PR. -->

 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [x] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

RKStrand committed:
Correct of Conduction Report Variable Issue

Corrected various problems with report variables relating to inside and outside face conduction variables.  Some variables simply were not set up to report.  Others were not listed in the docs.  Also did a name clean-up for consistency between similar values.

RKStrand committed:
Unit Test for 10812 Fix

Added a unit test to verify that the proper values are being reported for some of the variables that were added as part of this fix.

mjwitte commented on testfiles/1ZoneUncontrolled.idf:
```diff
@@ -400,6 +402,66 @@
 
   Output:Variable,*,Surface Outside Face Convection Heat Transfer Coefficient,daily;
 
+  Output:Variable,*,Surface Inside Face Conduction Heat Transfer Rate,hourly;
```
1ZoneUncontrolled has an rvi file in \testfiles. The new variables need to be added there as well.

RKStrand replied:
I can fix this when the crash issue you mentioned gets resolved.

RKStrand replied:
Had to make another commit because I forgot to do this before I committed the fix to resolve the initialization issue, but a revised .rvi is in the very latest commit.

mjwitte commented on tst/EnergyPlus/unit/HeatBalanceSurfaceManager.unit.cc:
```diff
@@ -8922,4 +8922,175 @@ TEST_F(EnergyPlusFixture, HeatBalanceSurfaceManager_SurroundingSurfacesTempTest)
     EXPECT_DOUBLE_EQ(srdSurfacesTemp_result, surface.SrdSurfTemp);
 }
 
+TEST_F(EnergyPlusFixture, HeatBalanceSurfaceManager_ZoneFaceConductionVariableTest)
```
This unit test crashes in `SolarShading::ReportSurfaceShading`. Using Debug build locally on windows.

RKStrand replied:
Ummm...well, I just tried this and it doesn't crash on Mac.  Do you know what line it is crashing on?

@Myoldmopar Any idea why it's fine on the Mac but not on Windows?  Not sure I've ever had this happen...or at least not recently.

mjwitte replied:
> Ummm...well, I just tried this and it doesn't crash on Mac. Do you know what line it is crashing on?
> 
> @Myoldmopar Any idea why it's fine on the Mac but not on Windows? Not sure I've ever had this happen...or at least not recently.

Sorry, forgot to paste in the screen shot of the error and line. Here:
<img width="1386" height="356" alt="image" src="https://github.com/user-attachments/assets/0502786e-6a1f-4ce7-a2cd-0fb5ddbfa289" />

`SurfSunlitFrac` is nan, probably never initialized. 

Why it doesn't crash on your Mac?
1. I've merged in develop.
2. In VS for windows, I've long ago had to manually activate exceptions for uninitialized variables and floating point exceptions.

mjwitte replied:
Or 3. Mac c++ might autoinitialize everything to zero?

RKStrand replied:
Huh, interesting.  Well, to avoid things relying on the OS, I just made a new commit that specifically initializes the Sunlit variables to zero.  Does this take care of the crash on Windows?

mjwitte replied:
> Huh, interesting. Well, to avoid things relying on the OS, I just made a new commit that specifically initializes the Sunlit variables to zero. Does this take care of the crash on Windows?

@RKStrand Thanks, that fixed the crash. I still need to review docs, then this should be good to go.

mjwitte reviewed (commented):
Tested with 1ZoneUncontrolled, the zone-level outputs agree with the sums of the surface-level outputs.

RKStrand committed:
Merge branch 'develop' into 10812ClarifyZoneOpaqueSurfaceConductionOutputVars

RKStrand committed:
Corrections for Strange Init Issue

In Windows, one of these was not being initialized and this resulted in a crash.  Zeroing them all out on definition in this unit test now to hopefully avoid this.

RKStrand committed:
Forgot the RVI

Another review comment involved the RVI for 1ZoneUncontrolled.  This resolves this comment hopefully.

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 10812ClarifyZoneOpaqueSurfaceConductionOutputVars

mjwitte committed:
Update I/O Ref to match source code names

mjwitte reviewed (approved):
@RKStrand I've updated some of the variable names in the I/O Ref to match what is in the source code. Please confirm these are correct: https://github.com/NREL/EnergyPlus/pull/11110/commits/1cf91aebe64b9881bb77da7a5de2a8428bee543b

Otherwise, this can merge once CI comes back clean.
